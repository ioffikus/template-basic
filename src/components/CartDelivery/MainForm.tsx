import React from 'react';
import { Spin, Form, Input, Row, Col, Select } from 'antd';
import { WrappedFormUtils } from 'antd/lib/form/Form';
import { IUserDeliveryState } from 'src/ducks/userDelivery';

interface IOwnProps {
  ui: Alicanto.Models.UI;
  initialValues: IUserDeliveryState;
  onSubmit: (values: IUserDeliveryState) => void;
  submitButtonRef: (submitButton: HTMLButtonElement) => void;
  isRequesting: boolean;
}

interface IFormProps {
  form: WrappedFormUtils;
}

type IProps = IOwnProps & IFormProps;

interface IState {}

const WrappedMainForm = Form.create<IOwnProps>()(
  class MainForm extends React.Component<IProps, IState> {
    handleSubmit = (event: React.SyntheticEvent<any>) => {
      event.preventDefault();
      this.props.form.validateFieldsAndScroll((error: any, values: any) => {
        if (!error) {
          this.props.onSubmit(values);
        }
      });
    };

    render() {
      const { getFieldDecorator } = this.props.form;
      const inputsUI = this.props.ui.routes.cart.userInfoForm.inputs;
      const validationUI = this.props.ui.common.forms.validation;
      return (
        <div className="root">
          <style jsx>{`
            .root {
              margin-bottom: 16px;
            }
            .root :global(.ant-select-selection__rendered) {
              margin-right: 24px;
              margin-left: 24px;
            }
            .submit-button {
              display: none;
            }
          `}</style>
          <Spin spinning={this.props.isRequesting}>
            <Form onSubmit={this.handleSubmit}>
              <button tabIndex={-1} type="submit" className="submit-button" ref={this.props.submitButtonRef} />
              <Row type="flex" gutter={16}>
                <Col span={12}>
                  <Form.Item>
                    {getFieldDecorator('name', {
                      initialValue: this.props.initialValues.name,
                      rules: [
                        {
                          required: true,
                          message: validationUI.notEmpty.i18n.defaultText,
                        },
                      ],
                    })(<Input size="large" placeholder={inputsUI.name.i18n.defaultText} />)}
                  </Form.Item>
                </Col>
                <Col span={12}>
                  <Form.Item>
                    {getFieldDecorator('surname', {
                      initialValue: this.props.initialValues.surname,
                      rules: [
                        {
                          required: true,
                          message: validationUI.notEmpty.i18n.defaultText,
                        },
                      ],
                    })(<Input size="large" placeholder={inputsUI.surname.i18n.defaultText} />)}
                  </Form.Item>
                </Col>
              </Row>
              <Row type="flex" gutter={16}>
                <Col span={12}>
                  <Form.Item>
                    {getFieldDecorator('email', {
                      initialValue: this.props.initialValues.email,
                      rules: [
                        {
                          type: 'email',
                          message: validationUI.email.i18n.defaultText,
                        },
                        {
                          required: true,
                          message: validationUI.notEmpty.i18n.defaultText,
                        },
                      ],
                    })(<Input size="large" placeholder={inputsUI.email.i18n.defaultText} />)}
                  </Form.Item>
                </Col>
                <Col span={12}>
                  <Form.Item>
                    {getFieldDecorator('phoneNumber', {
                      initialValue: this.props.initialValues.phoneNumber,
                      rules: [
                        {
                          required: true,
                          message: validationUI.notEmpty.i18n.defaultText,
                        },
                      ],
                    })(
                      <Input
                        addonBefore={getFieldDecorator('phonePrefix', {
                          initialValue: this.props.initialValues.phonePrefix,
                        })(
                          <Select>
                            <Select.Option value="7">+7</Select.Option>
                            <Select.Option value="380">+380</Select.Option>
                            <Select.Option value="375">+375</Select.Option>
                            <Select.Option value="371">+371</Select.Option>
                          </Select>,
                        )}
                        placeholder={inputsUI.phoneNumber.i18n.defaultText}
                      />,
                    )}
                  </Form.Item>
                </Col>
              </Row>
              <Row type="flex" gutter={16}>
                <Col span={12}>
                  <Form.Item>
                    {getFieldDecorator('country', {
                      initialValue: this.props.initialValues.country,
                      rules: [
                        {
                          required: true,
                          message: validationUI.notEmpty.i18n.defaultText,
                        },
                      ],
                    })(<Input size="large" placeholder={inputsUI.country.i18n.defaultText} />)}
                  </Form.Item>
                </Col>
                <Col span={12}>
                  <Form.Item>
                    {getFieldDecorator('stateRegion', {
                      initialValue: this.props.initialValues.stateRegion,
                      rules: [
                        {
                          required: true,
                          message: validationUI.notEmpty.i18n.defaultText,
                        },
                      ],
                    })(<Input size="large" placeholder={inputsUI.stateRegion.i18n.defaultText} />)}
                  </Form.Item>
                </Col>
              </Row>
              <Row type="flex" gutter={16}>
                <Col span={6}>
                  <Form.Item>
                    {getFieldDecorator('zipCode', {
                      initialValue: this.props.initialValues.zipCode,
                      rules: [
                        {
                          required: true,
                          message: validationUI.notEmpty.i18n.defaultText,
                        },
                      ],
                    })(<Input size="large" placeholder={inputsUI.zipCode.i18n.defaultText} />)}
                  </Form.Item>
                </Col>
                <Col span={18}>
                  <Form.Item>
                    {getFieldDecorator('address', {
                      initialValue: this.props.initialValues.address,
                      rules: [
                        {
                          required: true,
                          message: validationUI.notEmpty.i18n.defaultText,
                        },
                      ],
                    })(<Input size="large" placeholder={inputsUI.address.i18n.defaultText} />)}
                  </Form.Item>
                </Col>
              </Row>
            </Form>
          </Spin>
        </div>
      );
    }
  },
);

export default WrappedMainForm;
