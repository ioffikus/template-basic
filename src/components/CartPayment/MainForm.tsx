import React from 'react';
import { Spin, Row, Col, Form, Input } from 'antd';
import { WrappedFormUtils } from 'antd/lib/form/Form';

interface IOwnProps {
  ui: Alicanto.Models.UI;
  onSubmit: (values: any) => void;
  submitButtonRef: (submitButton: HTMLButtonElement) => void;
  isRequesting: boolean;
}

interface IFormProps {
  form: WrappedFormUtils;
}

type IProps = IOwnProps & IFormProps;

interface IState {}

const WrappedMainForm = Form.create<IOwnProps>()(
  class MainForm extends React.Component<IProps, IState> {
    handleSubmit = (event: React.SyntheticEvent<any>) => {
      event.preventDefault();
      this.props.form.validateFieldsAndScroll((error: any, values: any) => {
        if (!error) {
          console.info('Payment done');
          this.props.onSubmit(values);
        }
      });
    };

    render() {
      const { getFieldDecorator } = this.props.form;
      const inputsUI = this.props.ui.routes.cart.paymentForm.inputs;
      const validationUI = this.props.ui.common.forms.validation;
      return (
        <div className="root">
          <style jsx>{`
            .cart-info {
              display: flex;
              padding-bottom: 10px;
              margin-bottom: 21px;
            }
            .cart {
              position: relative;
              flex-basis: 50%;
              padding: 32px;
              border-radius: 32px;
              background-color: #ffffff;
            }
            .front {
              transform: translateX(25%);
              z-index: 2;
              box-shadow: 0 2px 16px 0 rgba(36, 103, 164, 0.18);
            }
            .back {
              transform: translateX(-25%);
              z-index: 1;
              display: flex;
              align-items: flex-end;
            }
            .cart-inner {
              width: 100%;
            }
            .root :global(.no-margin-bottom) {
              margin-bottom: 0;
            }
            .label {
              padding-top: 14px;
              color: #a7b4c3;
              font-size: 14px;
              text-transform: uppercase;
            }
            .root :global(.ant-input::placeholder) {
              text-transform: uppercase;
            }
            .submit-button {
              display: none;
            }
          `}</style>
          <Spin spinning={this.props.isRequesting}>
            <Form onSubmit={this.handleSubmit}>
              <button tabIndex={-1} type="submit" className="submit-button" ref={this.props.submitButtonRef} />
              <div className="cart-info">
                <div className="cart front">
                  <Row>
                    <Form.Item>
                      {getFieldDecorator('cartNumber', {
                        rules: [
                          {
                            required: true,
                            message: validationUI.notEmpty.i18n.defaultText,
                          },
                        ],
                      })(<Input size="large" placeholder={inputsUI.cardNumber.i18n.defaultText} />)}
                    </Form.Item>
                  </Row>
                  <Row>
                    <Form.Item>
                      {getFieldDecorator('cardholder', {
                        rules: [
                          {
                            required: true,
                            message: validationUI.cardNumber.i18n.defaultText,
                          },
                        ],
                      })(<Input size="large" placeholder={inputsUI.cardholderName.i18n.defaultText} />)}
                    </Form.Item>
                  </Row>
                  <Row type="flex" justify="end" gutter={16}>
                    <Col>
                      <div className="label">{this.props.ui.routes.cart.paymentForm.label.i18n.defaultText}</div>
                    </Col>
                    <Col span={7}>
                      <Form.Item className="no-margin-bottom">
                        {getFieldDecorator('cartMonth', {
                          rules: [
                            {
                              required: true,
                              message: validationUI.expDate.i18n.defaultText,
                            },
                          ],
                        })(<Input size="large" placeholder={inputsUI.expDatePlaceholderMonth.i18n.defaultText} />)}
                      </Form.Item>
                    </Col>
                    <Col span={7}>
                      <Form.Item className="no-margin-bottom">
                        {getFieldDecorator('cartYear', {
                          rules: [
                            {
                              required: true,
                              message: validationUI.expDate.i18n.defaultText,
                            },
                          ],
                        })(<Input size="large" placeholder={inputsUI.expDatePlaceholderYear.i18n.defaultText} />)}
                      </Form.Item>
                    </Col>
                  </Row>
                </div>
                <div className="cart back">
                  <div className="cart-inner">
                    <Row type="flex" justify="end" gutter={16} align="bottom">
                      <Col span={7}>
                        <Form.Item className="no-margin-bottom">
                          {getFieldDecorator('csv', {
                            rules: [
                              {
                                required: true,
                                message: validationUI.notEmpty.i18n.defaultText,
                              },
                            ],
                          })(<Input size="large" placeholder={inputsUI.securityCode.i18n.defaultText} />)}
                        </Form.Item>
                      </Col>
                    </Row>
                  </div>
                </div>
              </div>
            </Form>
          </Spin>
        </div>
      );
    }
  },
);

export default WrappedMainForm;
