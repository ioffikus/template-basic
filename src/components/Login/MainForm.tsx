import React from 'react';
import { Spin, Form, Icon, Input, Button } from 'antd';
import { IObject } from 'src/core/interfaces/IObject';
import { WrappedFormUtils } from 'antd/lib/form/Form';

interface IOwnProps {
  onSubmit: ({ username, password }: { username: string; password: string }) => void;
  ui: Alicanto.Models.UI;
  isRequesting: boolean;
}

interface IFormProps {
  form: WrappedFormUtils;
}

type IProps = IOwnProps & IFormProps;

interface IState {}

const WrappedMainForm = Form.create<IOwnProps>()(
  class MainForm extends React.Component<IProps, IState> {
    componentDidMount() {
      // To disabled submit button at the beginning.
      this.props.form.validateFields(null);
    }

    hasErrors = (fieldsError: IObject<any>) => {
      return Object.keys(fieldsError).some(field => fieldsError[field]);
    };

    handleSubmit = (event: React.SyntheticEvent<any>) => {
      event.preventDefault();
      this.props.form.validateFields((error: any, values: any) => {
        if (!error) {
          this.props.onSubmit(values);
        }
      });
    };

    render() {
      const validationUI = this.props.ui.common.forms.validation;
      const loginFormUI = this.props.ui.routes.login.form;
      const { getFieldDecorator, getFieldsError, getFieldError, isFieldTouched } = this.props.form;

      // Only show error after a field is touched.
      const usernameError = isFieldTouched('username') && getFieldError('username');
      const passwordError = isFieldTouched('password') && getFieldError('password');
      return (
        <Spin spinning={this.props.isRequesting}>
          <Form onSubmit={this.handleSubmit}>
            <Form.Item validateStatus={usernameError ? 'error' : null} help={usernameError || ''}>
              {getFieldDecorator('username', {
                rules: [{ required: true, message: validationUI.notEmpty.i18n.defaultText }],
              })(
                <Input
                  prefix={<Icon type="user" style={{ fontSize: 13 }} />}
                  placeholder={loginFormUI.labels.username.i18n.defaultText}
                />,
              )}
            </Form.Item>
            <Form.Item validateStatus={passwordError ? 'error' : null} help={passwordError || ''}>
              {getFieldDecorator('password', {
                rules: [{ required: true, message: validationUI.notEmpty.i18n.defaultText }],
              })(
                <Input
                  prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
                  type="password"
                  placeholder={loginFormUI.labels.password.i18n.defaultText}
                />,
              )}
            </Form.Item>
            <Form.Item>
              <Button type="primary" htmlType="submit" disabled={this.hasErrors(getFieldsError())}>
                {loginFormUI.buttons.signIn.i18n.defaultText}
              </Button>
            </Form.Item>
          </Form>
        </Spin>
      );
    }
  },
);

export default WrappedMainForm;
