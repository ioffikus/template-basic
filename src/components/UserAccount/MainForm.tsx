import React from 'react';
import { Spin, Form, Col, Row, Input, Button } from 'antd';
import cx from 'classnames';
import { WrappedFormUtils } from 'antd/lib/form/Form';

interface IOwnProps {
  ui: Alicanto.Models.UI;
  user: Alicanto.Models.User;
  onSubmit: (user: Alicanto.Models.User, cb: () => void) => void;
  isRequesting: boolean;
}

interface IFormProps {
  form: WrappedFormUtils;
}

type IProps = IOwnProps & IFormProps;

interface IState {
  isEditMode: boolean;
}

const WrappedMainForm = Form.create<IOwnProps>()(
  class MainForm extends React.Component<IProps, IState> {
    state: IState = {
      isEditMode: false,
    };

    toggleEditMode = () => {
      this.setState((prevState, props) => {
        return {
          isEditMode: !prevState.isEditMode,
        };
      });
    };

    handleSubmit = (event: React.SyntheticEvent<any>) => {
      event.preventDefault();
      this.props.form.validateFieldsAndScroll((error: any, values: any) => {
        if (!error) {
          const { firstName, lastName } = values;
          const user = {
            email: this.props.user.email,
            first_name: firstName,
            last_name: lastName,
          };
          this.props.onSubmit(user, () => {
            this.setState({
              isEditMode: false,
            });
          });
        }
      });
    };

    render() {
      const { getFieldDecorator } = this.props.form;
      const userAccountUI = this.props.ui.routes.user.account;
      const validationUI = this.props.ui.common.forms.validation;
      const { email, first_name: firstName, last_name: lastName } = this.props.user;
      const formItemLayout = {
        labelCol: { span: 7 },
        wrapperCol: { span: 17 },
      };
      return (
        <div
          className={cx('root', {
            'edit-mode': this.state.isEditMode,
          })}
        >
          <style jsx>{`
            .root :global(.ant-form-item-label) {
              text-align: left;
              font-size: 11px;
              text-transform: uppercase;
            }
            .root :global(label) {
              color: #a7b4c3;
            }
            .root :global(.ant-form-item-required:before),
            .root :global(.ant-input),
            .edit-mode :global(.ant-form-text:not(.text-email)) {
              display: none;
            }
            .edit-mode :global(.ant-input) {
              display: block;
            }
            .button-wrapper {
              text-align: right;
            }
            .root :global(.button-save) {
              display: block;
              width: 100%;
            }
          `}</style>
          <Row type="flex" gutter={16} align="middle" className="item">
            <Col span={11}>
              <Spin spinning={this.props.isRequesting}>
                <Form onSubmit={this.handleSubmit}>
                  <div className="button-wrapper">
                    <Button type="primary" icon="edit" onClick={this.toggleEditMode}>
                      {userAccountUI.button.i18n.defaultText}
                    </Button>
                  </div>
                  <Form.Item {...formItemLayout} colon={false} label={userAccountUI.form.labels.email.i18n.defaultText}>
                    <span className="ant-form-text text-email">{email}</span>
                  </Form.Item>
                  <Form.Item {...formItemLayout} colon={false} label={userAccountUI.form.labels.name.i18n.defaultText}>
                    {getFieldDecorator('firstName', {
                      rules: [{ required: true, message: validationUI.notEmpty.i18n.defaultText }],
                      initialValue: firstName,
                    })(<Input />)}
                    <span className="ant-form-text">{firstName}</span>
                  </Form.Item>
                  <Form.Item
                    {...formItemLayout}
                    colon={false}
                    label={userAccountUI.form.labels.surname.i18n.defaultText}
                  >
                    {getFieldDecorator('lastName', {
                      rules: [{ required: true, message: validationUI.notEmpty.i18n.defaultText }],
                      initialValue: lastName,
                    })(<Input />)}
                    <span className="ant-form-text">{lastName}</span>
                  </Form.Item>
                  {this.state.isEditMode && (
                    <Form.Item>
                      <Button type="primary" icon="save" size="large" className="button-save" htmlType="submit">
                        {userAccountUI.form.submit.i18n.defaultText}
                      </Button>
                    </Form.Item>
                  )}
                </Form>
              </Spin>
            </Col>
          </Row>
        </div>
      );
    }
  },
);

export default WrappedMainForm;
