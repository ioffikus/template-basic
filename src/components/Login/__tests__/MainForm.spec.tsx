import React from 'react';
import { shallow } from 'enzyme';
import { Form, Input, Button, Spin } from 'antd';
import uiStateMock from 'src/store/__mocks__/uiStore';
import MainForm from '../MainForm';

describe(null, () => {
  console.warn = jest.fn();
  it('render()', () => {
    const dummyWrapper = shallow(<MainForm onSubmit={() => {}} ui={uiStateMock} isRequesting={false} />).dive();
    expect(dummyWrapper.dive()).toMatchSnapshot();
  });
  it('componentDidMount() should be call props.form.validateFields(null)', () => {
    const form = {
      validateFields: jest.fn(),
      getFieldDecorator: () => jest.fn(),
      getFieldsError: () => ({}),
      getFieldError: jest.fn(),
      isFieldTouched: jest.fn(),
    };
    const wrapper = shallow(<MainForm onSubmit={() => {}} ui={uiStateMock} isRequesting={false} />);
    const props = wrapper.props();
    wrapper.setProps({ ...props, form });
    wrapper.dive().dive(); // double dive() if wrapped with Form.create
    expect(form.validateFields).toBeCalledWith(null);
  });
  it('Form.onSubmit() without values', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<MainForm onSubmit={onSubmit} ui={uiStateMock} isRequesting={false} />);
    const dummyWrapper = wrapper.dive().dive(); // double dive() if wrapped with Form.create
    const formInstance = dummyWrapper
      .find(Form)
      .dive()
      .instance() as Form;
    const submitEvent = {
      preventDefault: jest.fn(),
    };
    formInstance.props.onSubmit(submitEvent as any);
    expect(submitEvent.preventDefault).toBeCalled();
    expect(onSubmit).not.toBeCalled();
  });
  it('Form.onSubmit() with values', () => {
    const values = { username: 'dummy username', password: 'dummy password' };
    const onSubmit = jest.fn();
    const wrapper = shallow(<MainForm onSubmit={onSubmit} ui={uiStateMock} isRequesting={false} />);
    const dummyWrapper = wrapper.dive().dive(); // double dive() if wrapped with Form.create
    const usernameInstance = dummyWrapper
      .find('Input#username')
      .dive()
      .instance() as Input;
    usernameInstance.props.onChange({ target: { value: values.username } } as any);
    const passwordInstance = dummyWrapper
      .find('Input#password')
      .dive()
      .instance() as Input;
    passwordInstance.props.onChange({ target: { value: values.password } } as any);
    const formInstance = dummyWrapper
      .find(Form)
      .dive()
      .instance() as Form;
    const submitEvent = {
      preventDefault: jest.fn(),
    };
    formInstance.props.onSubmit(submitEvent as any);
    expect(submitEvent.preventDefault).toBeCalled();
    expect(onSubmit).toBeCalledWith(values);
  });
  it('Button.disabled when Form has error', () => {
    const onSubmit = jest.fn();
    const form = {
      validateFields: jest.fn(),
      getFieldDecorator: () => jest.fn(),
      getFieldsError: () => ({ username: {} }),
      getFieldError: jest.fn(),
      isFieldTouched: jest.fn(),
    };
    const wrapper = shallow(<MainForm onSubmit={onSubmit} ui={uiStateMock} isRequesting={false} />);
    const props = wrapper.props();
    wrapper.setProps({ ...props, form });
    const dummyWrapper = wrapper.dive().dive(); // double dive() if wrapped with Form.create
    const button = dummyWrapper.find(Button);
    const buttonInstance = button.dive().instance() as Button;
    expect(buttonInstance.props.disabled).toBeTruthy();
    button.simulate('click'); // used simulate, because Button doesn't has onClick and buttonInstance doesn't has onClick method too
    expect(onSubmit).not.toBeCalled();
  });
  it('Spin.spinning when isRequesting', () => {
    const wrapper = shallow(<MainForm onSubmit={() => {}} ui={uiStateMock} isRequesting={true} />);
    const dummyWrapper = wrapper.dive().dive(); // double dive() if wrapped with Form.create
    const spinInstance = dummyWrapper
      .find(Spin)
      .dive()
      .instance() as Spin;
    expect(spinInstance.props.spinning).toBeTruthy();
  });
});
