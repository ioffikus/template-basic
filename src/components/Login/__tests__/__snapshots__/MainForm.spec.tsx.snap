// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() 1`] = `
<Spin
  prefixCls="ant-spin"
  size="default"
  spinning={false}
  wrapperClassName=""
>
  <Form
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
    prefixCls="ant-form"
  >
    <FormItem
      colon={true}
      hasFeedback={false}
      help=""
      prefixCls="ant-form"
      validateStatus={null}
    >
      <Input
        data-__meta={
          Object {
            "leadingName": "username",
            "name": "username",
            "originalProps": Object {
              "disabled": false,
              "placeholder": "Username",
              "prefix": <Icon
                style={
                  Object {
                    "fontSize": 13,
                  }
                }
                type="user"
              />,
              "prefixCls": "ant-input",
              "type": "text",
            },
            "ref": null,
            "rules": Array [
              Object {
                "message": "Please fill this field",
                "required": true,
              },
            ],
            "trigger": "onChange",
            "validate": Array [
              Object {
                "rules": Array [
                  Object {
                    "message": "Please fill this field",
                    "required": true,
                  },
                ],
                "trigger": Array [
                  "onChange",
                ],
              },
            ],
            "valuePropName": "value",
          }
        }
        disabled={false}
        id="username"
        onChange={[Function]}
        placeholder="Username"
        prefix={
          <Icon
            style={
              Object {
                "fontSize": 13,
              }
            }
            type="user"
          />
        }
        prefixCls="ant-input"
        type="text"
      />
    </FormItem>
    <FormItem
      colon={true}
      hasFeedback={false}
      help=""
      prefixCls="ant-form"
      validateStatus={null}
    >
      <Input
        data-__meta={
          Object {
            "leadingName": "password",
            "name": "password",
            "originalProps": Object {
              "disabled": false,
              "placeholder": "Password",
              "prefix": <Icon
                style={
                  Object {
                    "fontSize": 13,
                  }
                }
                type="lock"
              />,
              "prefixCls": "ant-input",
              "type": "password",
            },
            "ref": null,
            "rules": Array [
              Object {
                "message": "Please fill this field",
                "required": true,
              },
            ],
            "trigger": "onChange",
            "validate": Array [
              Object {
                "rules": Array [
                  Object {
                    "message": "Please fill this field",
                    "required": true,
                  },
                ],
                "trigger": Array [
                  "onChange",
                ],
              },
            ],
            "valuePropName": "value",
          }
        }
        disabled={false}
        id="password"
        onChange={[Function]}
        placeholder="Password"
        prefix={
          <Icon
            style={
              Object {
                "fontSize": 13,
              }
            }
            type="lock"
          />
        }
        prefixCls="ant-input"
        type="password"
      />
    </FormItem>
    <FormItem
      colon={true}
      hasFeedback={false}
      prefixCls="ant-form"
    >
      <Button
        disabled={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        prefixCls="ant-btn"
        type="primary"
      >
        Sign in
      </Button>
    </FormItem>
  </Form>
</Spin>
`;
